### PR TITLE
Implementing .whereLike and .whereILike api

### PR DESCRIPTION
**Pull request by LucasHFS**
Attempt to solve #2265 .

### CODE REVIEW

**Comment by kibertoad** [comment]
Thank you! Could you also please add some integration tests that actually retrieve some data? see `integration2` directory for good examples.

File: lib/dialects/mssql/query/mssql-querycompiler.js [hunk 1]
```diff
@@ -479,6 +479,22 @@ class QueryCompiler_MSSQL extends QueryCompiler {
     }
     return offset;
   }
+
+  whereLike(statement) {
+    return `[${
```
Comment (kibertoad) [inline lib/dialects/mssql/query/mssql-querycompiler.js:484]: Shouldn't this need some escaping for values? I think for columns we typically use `formatter.columnize`, and for a single value `client.parameter`. See `C:\sources\node\knex\lib\query\querycompiler.js` for an example.

File: lib/dialects/postgres/query/pg-querycompiler.js [hunk 2]
```diff
@@ -218,6 +218,18 @@ class QueryCompiler_PG extends QueryCompiler {
   distinctOn(value) {
     return 'distinct on (' + this.formatter.columnize(value) + ') ';
   }
+
+  whereLike(statement) {
```
Comment (kibertoad) [inline lib/dialects/postgres/query/pg-querycompiler.js:222]: Do we still need client-specific implementations of this functions if we use standard knex escaping mechanism? I assume there are no other differences?

**Comment by OlivierCavadenti** [comment]
Superseded by https://github.com/knex/knex/pull/4779

**Comment by xpuu** [comment]
Hey @OlivierCavadenti , thanks for your effort! What about orWhereLike and orWhereILike?

**Pull request closed without merging**